lib/src/main/java/graphql/nadel/validation/NadelHydrationArgumentValidation.kt

```diff
@@ -0,0 +1,288 @@
+                            actorFieldName
+                    )
+                }
+                return null
+            }
+            // if enums they must be the same
+            else if (unwrappedHydrationSourceFieldType is GraphQLEnumType && unwrappedActorFieldArgType is GraphQLEnumType){
+                if (unwrappedHydrationSourceFieldType.name != unwrappedActorFieldArgType.name) {
+                    return NadelSchemaValidationError.IncompatibleHydrationArgumentType(
+                            parent,
+                            overallField,
+                            remoteArg,
+                            hydrationSourceFieldType,
+                            actorFieldArgType,
+                            actorFieldName
+                    )
+                }
+                return null
+            }
+            // Any other types are not assignable
+            return NadelSchemaValidationError.IncompatibleHydrationArgumentType(
+                    parent,
+                    overallField,
+                    remoteArg,
+                    hydrationSourceFieldType,
+                    actorFieldArgType,
+                    actorFieldName
+            )
+        }
+
+        private fun validateScalarArg(
+                hydrationSourceFieldType: GraphQLType,
+                actorFieldArgType: GraphQLType,
+                parent: NadelServiceSchemaElement,
+                overallField: GraphQLFieldDefinition,
+                remoteArg: RemoteArgumentDefinition,
+                actorFieldName: String
+        ): NadelSchemaValidationError? {
+            if (hydrationSourceFieldType.unwrapNonNull() is GraphQLScalarType && actorFieldArgType.unwrapNonNull() is GraphQLScalarType) {
+                if (isScalarAssignable(hydrationSourceFieldType.unwrapNonNull() as GraphQLScalarType, actorFieldArgType.unwrapNonNull() as GraphQLScalarType)) {
+                    return null
+                } else {
+                    return NadelSchemaValidationError.IncompatibleHydrationArgumentType(
+                            parent,
+                            overallField,
+                            remoteArg,
+                            hydrationSourceFieldType,
+                            actorFieldArgType,
+                            actorFieldName
+                    )
+                }
+            }
+            return null
+        }
+
+        private fun isScalarAssignable(typeToAssign: GraphQLNamedInputType, targetType: GraphQLNamedInputType): Boolean {
+            if (typeToAssign.name == targetType.name) {
+                return true
+            }
+            if (targetType.name == Scalars.GraphQLID.name &&
+                    (typeToAssign.name == Scalars.GraphQLString.name || typeToAssign.name == Scalars.GraphQLInt.name)) {
+                return true
+            }
+            if (targetType.name == Scalars.GraphQLString.name && typeToAssign.name == Scalars.GraphQLID.name) {
```

Comment:
Allowing ID -> String. see comment here: https://hello.atlassian.net/wiki/spaces/PSRV/pages/2987269760/Analysis+of++Current+Hydration+Arg+Validation+Errors+in+Central+Schema?focusedCommentId=2990608894

Reply:
Not sure I agree with this.

Reply:
Sure, we'll have a discussion on the page